DataGrid: Rework the headerFilter.dataSource article (#747)
DevExpress/devextreme-documentation: api-reference/_hidden/GridBaseColumn/headerFilter/dataSource.md

---
---
##### shortDescription

> Specifies the header filter data source.

Reviewer: ```suggestion
Specifies the header filter's data source.
```

The widget's instance.

##### field(options.dataSource): DataSource_Options
A [DataSource](/Documentation/ApiReference/Data_Layer/DataSource/) instance.

---

> The **{WidgetName}** generates a header filter data source automatically based on column values. Use the **dataSource** option to change the generated data source or specify a custom data source.

Reviewer: ```suggestion
The **{WidgetName}** generates a header filter's data source automatically based on column values. Use the **dataSource** option to change the generated data source or specify a custom data source.
```

### Specify a Custom Data Source

> To define a data source from scratch, set the **dataSource** option to an array of objects. Each object configures one header filter item and should have the following fields:

Reviewer: "To define a data source from scratch" Can't you just replace it with 'to define a data source'? 

```suggestion
To define a data source, set the **dataSource** option to an array of objects. Each object configures one header filter item and should have the following fields:
```

                    text: 'More than $3000',
                    value: ['SaleAmount', '>', 3000]
                }]
            };
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { {WidgetName}, Column, HeaderFilter } from 'devextreme-react/{widget-name}';

    const headerFilterData = [{
        text: 'Zero',
        value: 0
    }, {
        text: 'Less than $3000',
        value: ['SaleAmount', '<', 3000]
    }, {
        text: 'More than $3000',
        value: ['SaleAmount', '>', 3000]
    }];

    class App extends React.Component {
        render() {
            return (
                <{WidgetName}>
                    <Column>
                        <HeaderFilter
                            dataSource={headerFilterData}
                        />
                    </Column>
                </{WidgetName}>
            );
        }
    }

    export default App;

---

#include common-demobutton with {
    url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/Filtering/"
}

### Map Data Source Fields

> Header filter data objects should have the `text` and `value` fields, but when the data objects are fetched from a server, you have no control over their fields. In this case, map the original data source to the `text` + `value` structure. A mapped data source should also include key fields from the original data source.

Reviewer: ```suggestion
Header filter data objects should have the `text` and `value` fields. However, data objects fetched from a server have fields assigned to them automatically. In this case, map the original data source to the `text` + `value` structure. A mapped data source should also include key fields from the original data source.
```

Author: ```suggestion
Header filter data objects should have the `text` and `value` fields. However, data objects fetched from a server may not have these fields. In this case, map the original data source to the `text` + `value` structure. A mapped data source should also include key fields from the original data source.
```

Reviewer: Ok

    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { {WidgetName}, Column, HeaderFilter } from 'devextreme-react/{widget-name}';

    class App extends React.Component {
        render() {
            return (
                <{WidgetName}>
                    <Column>
                        <HeaderFilter
                            dataSource={this.customizeHeaderFilterData}
                        />
                    </Column>
                </{WidgetName}>
            );
        }

        customizeHeaderFilterData(options) {
            options.dataSource.postProcess = function(results) {
                results.push({
                    text: 'Weekends',
                    value: [
                        ['OrderDay', '=', 0],
                            'or',
                        ['OrderDay', '=', 6]
                    ]
                });
                return results;
            };
        }
    }

    export default App;

---

#include common-demobutton with {
    url: "https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/Filtering/"
}

### Customize Item Appearance

> You can use [templates](/Documentation/ApiReference/Common/Object_Structures/template/) to customize the appearance of individual items. In the following code, a template applies the bold font to the `Today` item:

Reviewer: ```suggestion
You can use [templates](/Documentation/ApiReference/Common/Object_Structures/template/) to customize the appearance of individual items. In the following code, a template makes the `Today` item bold:
```
